presto-tests/src/test/java/io/prestosql/tests/TestInformationSchemaConnector.java

```diff
@@ -192,13 +196,20 @@ public void testMetadataCalls()
                         .withListSchemasCount(1)
                         .withListTablesCount(2)
                         .withGetColumnsCount(10000));
+        assertNoMetadataCalls("SELECT count(*) from test_catalog.information_schema.tables WHERE table_schema = ''", "VALUES 0");
     }
 
     @Test
     public void testInformationForEmptyNames()
     {
-        assertQuery("SELECT count(*) FROM test_catalog.information_schema.tables WHERE table_schema='tpch' AND table_name=''", "VALUES 0");
-        assertQuery("SELECT count(*) FROM test_catalog.information_schema.tables WHERE table_schema='' AND table_name='table'", "VALUES 0");
+        assertFalse(searchFrom(getQueryRunner().createPlan(getSession(), "SELECT count(*) from test_catalog.information_schema.tables WHERE table_schema = ''", WarningCollector.NOOP).getRoot())
+                .where(TableScanNode.class::isInstance)
+                .findFirst()
+                .isPresent());
+        assertFalse(searchFrom(getQueryRunner().createPlan(getSession(), "SELECT count(*) from test_catalog.information_schema.tables WHERE table_name = ''", WarningCollector.NOOP).getRoot())
+                .where(TableScanNode.class::isInstance)
+                .findFirst()
+                .isPresent());
```

Comment:
please extract a method `assertNoTableScan("SELECT count(*) from test_catalog.information_schema.tables WHERE table_name = ''")`

Also please pass some message to `assertFalse`